### PR TITLE
Task/fp 1499 ui pattern demo

### PR DESCRIPTION
**Pull request by wesleyboar**
## Overview

Add UI Pattern Demo using [Fractal].

[fractal]: https://fractal.build/

### To Do

- [x] [no italics for button icons](https://github.com/TACC/Core-CMS/pull/527#issuecomment-1216793576) — https://github.com/TACC/Core-Styles/commit/474eff326b925e5ed246f9f18cca9ef477408a7a

## Related

- [FP-1499](https://jira.tacc.utexas.edu/browse/FP-1499)
- ~~requires https://github.com/TACC/tup-ui/pull/45~~
   requires https://github.com/TACC/Core-Styles/pull/34

## Changes

- gitignore & docs
- fractal config & theme
- symlinks to gather assets

## Testing

1. `npm ci`
2. `npm run build --project=core-cms`\
    <sub>(**requires** a project e.g. `core-cms`, `ecep-cms`, `frontera-cms`)</sub>
3. `npm run start:css-demo`
4. Open http://localhost:3000/.
5. In sidebar, drill down into PATTERNS → Core → Components → ….
6. Review variations and styles.

Command name suggestions are welcome.

## UI

No remote server available. Pending new architecture.

<details><summary>Screenshots</summary>

| ecep-cms | core-cms |
| - | - |
| ecep, primary | cep, primary |
ecep, secondary | cep, secondary |
| ecep, tertiary | cep, tertiary |
| ecep, small | cep, small |

</details>

### Fixes

<details><summary>no italics for pseudo-icons (<a href="https://github.com/TACC/Core-Styles/commit/474eff326b925e5ed246f9f18cca9ef477408a7a">TACC/Core-Styles@<code>474eff3</code></a>)</summary>

fp-1499 474eff3 core-cms

</details>

### Known Issues

<details><summary>1.  When we have a way to deploy, we might drop the ability to support building per project…</summary>

…because the symlink-ed `styles/projects` directory includes many irrelevant files (`.py`, `.md`, `.html`).

The feature could come back if and after Core-CMS-Resources become their own containerized projects.

</details>

### CODE REVIEW

File: taccsite_ui/fractal.config.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// Set website paths
+/*
+fractal.web.set('static.path',
+  path.dirname( require.resolve('@tacc/core-styles')) +
+  '/..' + // exits 'src/' which is at the end of what require.resolve returns
+  '/dist'
+);
+*/
+fractal.web.set('static.path', __dirname + '/styles');
```
Comment (wesleyboar) [inline taccsite_ui/fractal.config.js:54]: This is directory that just has symlinks. Fractal can not read from more than one directory.

[known issue (at "Supporting multiple component source directories.")](https://github.com/frctl/fractal/issues/197)

File: taccsite_ui/fractal.config.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// (stylesheet)
+const cmsCSSFile = 'cms/site.css';
+const projCSSFile = ( projName )
+  ? `projects/${projName}/static/${projName}/css/build/site.css`
+  : null;
+
+// Set source paths
+// (for components)
+fractal.components.set('exclude', '*.md');
+fractal.components.set('path', __dirname + '/patterns');
```
Comment (wesleyboar) [inline taccsite_ui/fractal.config.js:26]: This is directory that just has symlinks. Fractal can not read from more than one directory.

[known issue](https://github.com/frctl/fractal/issues/207)

**Review by wesleyboar — COMMENTED** [review]
The future intent—if we still don't have storybook yet (pending a snag in tup-ui using it for components)—is that a Dockerfile is added so this can be hosted properly as a [custom app](https://github.com/TACC/Core-CMS/pull/517).

**Review by rstijerina — APPROVED** [review]
LGTM, nice work. Any reason why the close x's look italicized?
Screen Shot 2022-08-16 at 12 48 03 AM

**Comment by wesleyboar** [comment]
> LGTM, nice work. Any reason why the close x's look italicized? Screen Shot 2022-08-16 at 12 48 03 AM

@rstijerina, yes:
1. I used an `<i>…</i>` tag for the icon.
2. I did not use a Cortal icon class nor add Cortal icon support.
3. The default browser style for an `<i>…</i>` tag is italics.

When I add cortal icons to Core-Styles, I can make that a Cortal icon. Alternatively—to truly isolate modules (not mix cortal icons into button)—I could turn off the italic style for just that pattern and only in the pattern demo. Hm, I like that. Be right back.

**Comment by wesleyboar** [comment]
Fixed italics on pseudo-icons:
fp-1499 474eff3 core-cms

**Pull request merged**